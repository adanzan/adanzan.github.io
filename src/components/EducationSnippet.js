import * as React from 'react'

import styles from '../styles/Snippet.module.css'

import { Paper } from '@mui/material'

export default function EducationSnippet ({ name, dateRange, location, logo, link, gpa, achievements }) {
  const achievementElement = achievements.map((achievement, index) => {
    return (
            <li key={index}>
                {achievement}
            </li>
    )
  })

  return (
        <div className={styles.EducationSnippet}>
            <Paper className={styles.snippetPaper} elevation={+10} style={{ backgroundColor: '#fff3e0' }}>
                <div className={styles.org}>
                    <a href={link} target="_blank" rel="noopener noreferrer">
                        <img className={styles.logo} src={logo} alt={name + 'logo'}/>
                    </a>
                    <div className={styles.nameLocationWrapper}>
                        <div className={styles.nameLocation}>
                            <a className={styles.name} href={link} target="_blank" rel="noopener noreferrer">
                                <h3>{name}</h3>
                            </a>
                            <p className={styles.location}>{location}</p>
                        </div>
                    </div>
                    <div className={styles.detailsWrapper}>
                        <div className={styles.details}>
                            <h4>{dateRange}</h4>
                            <p> <strong>GPA:</strong> {gpa}</p>
                            <p>{achievementElement}</p>
                        </div>
                    </div>
                </div>
            </Paper>
        </div>
  )
}
