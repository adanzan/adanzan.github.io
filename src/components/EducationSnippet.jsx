/* eslint no-unused-vars: 0 */

import * as React from "react";
import PropTypes from "prop-types";
import { Paper } from "@mui/material";
import styles from "../styles/Snippet.module.css";

export default function EducationSnippet({
	name, degree, dateRange, location, logo, link, gpa, achievements,
}) {
	const achievementElement = achievements.map((achievement) => (
		<li key={achievement}>
			{achievement}
		</li>
	));

	return (
		<div className={styles.EducationSnippet}>
			<Paper className={styles.snippetPaper} elevation={+10}>
				<div className={styles.org}>
					<a href={link} target="_blank" rel="noopener noreferrer">
						<img className={styles.logo} src={logo} alt={`${name}logo`} />
					</a>
					<div className={styles.nameLocationWrapper}>
						<div className={styles.nameLocation}>
							<a className={styles.name} href={link} target="_blank" rel="noopener noreferrer">
								<h3>{name}</h3>
							</a>
							<p className={styles.location}>{location}</p>
						</div>
					</div>
				</div>
				<div className={styles.detailsWrapper}>
					<div className={styles.details}>
						<p className={styles.detailsDegree}>
							{" "}
							<strong>{degree}</strong>
							{" "}
						</p>
						<p className={styles.detailsDate}>{dateRange}</p>
						<p>
							{" "}
							<strong>GPA:</strong>
							{" "}
							{gpa}
						</p>
						<p>{achievementElement}</p>
					</div>
				</div>
			</Paper>
		</div>
	);
}

EducationSnippet.propTypes = {
	name: PropTypes.string.isRequired,
	degree: PropTypes.string.isRequired,
	dateRange: PropTypes.string.isRequired,
	location: PropTypes.string.isRequired,
	logo: PropTypes.element.isRequired,
	link: PropTypes.string.isRequired,
	gpa: PropTypes.string.isRequired,
	achievements: PropTypes.arrayOf(
		PropTypes.string,
	).isRequired,
};
