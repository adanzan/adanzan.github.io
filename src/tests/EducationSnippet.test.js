/* eslint no-unused-vars: 0 */
import * as React from "react";
import { render } from "@testing-library/react";

import EducationSnippet from "../components/EducationSnippet"; 

let testRender;

describe("EducationSnippet component", () => {
	beforeEach(() => {
		testRender = render(<EducationSnippet 
			name={"testName"}
			degree={"testDegree"}
			dateRange={"testDateRange"}
			location={"testLocation"}
			logo={"testLogo"}
			link={"testLink"}
			gpa={"testGpa"}
			achievements={["testAchievement1", "testAchievement2"]}
		/>);
	});
	test("matches snapshot", () => {
		expect(testRender).toMatchSnapshot();
	});

	describe("correctly displays", () => {
		test("name", () => {
			expect(testRender.getByRole("heading", "name")).toContainHTML("testName");
		});
		test("testGpa", () => {
			expect(testRender.getByText("testGpa"));
		});
		test("achievements", () => {
			expect(testRender.getByText("testAchievement1"));
			expect(testRender.getByText("testAchievement2"));
		});
	});
});

