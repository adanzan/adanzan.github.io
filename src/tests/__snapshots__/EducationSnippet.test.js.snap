// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EducationSnippet component matches snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="EducationSnippet"
      >
        <div
          class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation0 snippetPaper css-1khjo2-MuiPaper-root"
        >
          <div
            class="org"
          >
            <a
              href="testLink"
              rel="noopener noreferrer"
              target="_blank"
            >
              <img
                alt="testNamelogo"
                class="logo"
                src="testLogo"
              />
            </a>
            <div
              class="nameLocationWrapper"
            >
              <div
                class="nameLocation"
              >
                <a
                  class="name"
                  href="testLink"
                  rel="noopener noreferrer"
                  target="_blank"
                >
                  <h3>
                    testName
                  </h3>
                </a>
                <p
                  class="location"
                >
                  testLocation
                </p>
              </div>
            </div>
          </div>
          <div
            class="detailsWrapper"
          >
            <div
              class="details"
            >
              <p
                class="detailsDegree"
              >
                 
                <strong>
                  testDegree
                </strong>
                 
              </p>
              <p
                class="detailsDate"
              >
                testDateRange
              </p>
              <p>
                 
                <strong>
                  GPA:
                </strong>
                 
                testGpa
              </p>
              <p>
                <li>
                  testAchievement1
                </li>
                <li>
                  testAchievement2
                </li>
              </p>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="EducationSnippet"
    >
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation0 snippetPaper css-1khjo2-MuiPaper-root"
      >
        <div
          class="org"
        >
          <a
            href="testLink"
            rel="noopener noreferrer"
            target="_blank"
          >
            <img
              alt="testNamelogo"
              class="logo"
              src="testLogo"
            />
          </a>
          <div
            class="nameLocationWrapper"
          >
            <div
              class="nameLocation"
            >
              <a
                class="name"
                href="testLink"
                rel="noopener noreferrer"
                target="_blank"
              >
                <h3>
                  testName
                </h3>
              </a>
              <p
                class="location"
              >
                testLocation
              </p>
            </div>
          </div>
        </div>
        <div
          class="detailsWrapper"
        >
          <div
            class="details"
          >
            <p
              class="detailsDegree"
            >
               
              <strong>
                testDegree
              </strong>
               
            </p>
            <p
              class="detailsDate"
            >
              testDateRange
            </p>
            <p>
               
              <strong>
                GPA:
              </strong>
               
              testGpa
            </p>
            <p>
              <li>
                testAchievement1
              </li>
              <li>
                testAchievement2
              </li>
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
